e2e_tests/test_workspace_services.py

```diff
@@ -56,7 +56,7 @@ async def test_create_guacamole_service_into_base_workspace(admin_token, verify)
 
 @pytest.mark.extended_aad
 @pytest.mark.timeout(3000)
-async def test_create_guacamole_service_into_aad_workspace(admin_token, workspace_owner_token, verify) -> None:
+async def test_create_guacamole_service_into_aad_workspace(admin_token, verify) -> None:
```

Comment:
Why have you changed the token to the admin one? A workspace service needs to be deployed by the owner... 
Or, am I missing something?

Reply:
The E2E tests create the workspace dynamically, and to do that they need the admin token as that route is an admin only route. We used to have a fixture that would get the workspace token by using a convention `api://<app_reg_id`; that convention is not true anymore as we need to read a property of the workspace (scope_id) to tell us what that value is. Once we have that value, then we dynamically get the workspace_token scoped to that application and then use the workspace_owner token from then on in.

Basically, what used to be a test fixture, is now a method called **get_workspace_owner_token** that you call from your test.